lib/game.rb

```diff
@@ -45,7 +45,24 @@ def take_computer_turn
   end
 
   def bid_farewell
-    text_view.render_game_over
+    print_player_won
+    print_computer_won
+    print_tie
+  end
+
+  def print_player_won
+    return unless grid.player_won?
+    text_view.render_player_won
+  end
+
+  def print_computer_won
+    return unless grid.computer_won?
+    text_view.render_computer_won
+  end
+
+  def print_tie
+    return unless grid.full?
```

Comment:
Am I abusing guard clauses with these three methods?

Reply:
YES, and it's great. 

When you start to see this kind of symmetry. It is an indicator of missing objects, in fact three missing objects. You seem to have three printer classes and each of these classes knows whether or not it won and how to print its status. If you separate those responsibilities you end up being able to loop through the objects, choose he one that `won?` and then ask it to print the status.

For example this would be one class of three:
```ruby
PlayerStatus = Struct.new(:grid, :text_view) do
  def won?
     grid.player_won?
  end

  def print
    text_view.render_player_won
  end
end
```
Then in this base class you could say: `possible_statuses.detect(&:won).print`, where possible statuses are the instances generated from these three classes.

Moving that method might imply though that these class and not the grid can understand whether they won. But maybe not.

Reply:
Also ignore all the struct stuff. I just didn't want to create accessors and stuff.

Reply:
Oh, I really like that idea!